web/src/containers/QuickExchange/index.tsx

```diff
@@ -0,0 +1,373 @@
+    getCurrencyFiltred,
+    getMarket,
+    getWallets,
+    getWallet,
+    getBaseAmount,
+    getQuoteAmount,
+} from './helpers';
+import { precisionRegExp, ssToMMSS } from '../../helpers';
+import { ArrowRight, WarningIcon } from './icons';
+
+interface QuickExchangeTimer {
+    initialSeconds: number;
+    initialMinutes: number;
+};
+
+interface QuickExchangeInterface {
+    amount: string;
+    currency: string;
+};
+
+const DEFAULT_VALUE_TIMER: QuickExchangeTimer = {
+    initialSeconds: 0,
+    initialMinutes: 0,
+};
+
+const DEFAULT_VALUE: QuickExchangeInterface = {
+    amount: '',
+    currency: '',
+};
+
+export const QuickExchangeContainer = () => {
+    const [base, setBaseData] = useState<QuickExchangeInterface>(DEFAULT_VALUE);
+    const [quote, setQuoteData] = useState<QuickExchangeInterface>(DEFAULT_VALUE);
+    const [type, setType] = useState('buy');
+    const [dropdownKey, setDropdownKey] = useState('');
+    const [marketID, setMarket] = useState('');
+    const [basePrecision, setBasePrecision] = useState(0);
+    const [quotePrecision, setQuotePrecision] = useState(0);
+    const [time, setTime] = useState<QuickExchangeTimer>(DEFAULT_VALUE_TIMER);
+
+    const { formatMessage } = useIntl();
+    const dispatch = useDispatch();
+
+    const wallets = useSelector(selectWallets) || [];
+    const markets = useSelector(selectMarkets) || [];
+    const marketPrice = useSelector(selectMarketPrice);
+    const updateTimer = useSelector(selectMarketPriceFetchSuccess);
+
+    const translate = useCallback((id: string) => formatMessage({ id: id }), [formatMessage]);
+
+    useCurrenciesFetch();
+
+    const updateMarketPrice = () => {
+        if (currentSelectedMarket) {
+            dispatch(marketPriceFetch({ market: currentSelectedMarket.id, side: type }));
+        }
+    };
+
+    React.useEffect(() => {
+        const seconds = +msPricesUpdates() / 1000;
+        setTime(ssToMMSS(seconds));
+
+        dispatch(walletsFetch());
+        dispatch(marketsFetch({type: 'qe'}));
+    }, []);
+
+    const currentSelectedMarket = React.useMemo(() => getMarket(marketID, markets), [marketID]);
+
+    React.useEffect(() => {
+        updateMarketPrice();
+    }, [currentSelectedMarket, type]);
+
+    const marketCurrency = React.useMemo(() => getCurrencyForMarket(markets), [markets]);
+    const marketsBaseUnit = React.useMemo(() =>
+        getCurrencyFiltred(quote.currency, base.currency, dropdownKey, marketCurrency, markets), [base, quote, dropdownKey, marketCurrency, markets]);
+    const marketsQuoteUnit = React.useMemo(() =>
+        getCurrencyFiltred(base.currency, quote.currency, dropdownKey, marketCurrency, markets), [base, quote, dropdownKey, marketCurrency, markets]);
+    const walletsBaseUnit = React.useMemo(() =>
+        getWallets(wallets, marketsBaseUnit), [wallets, marketsBaseUnit]);
+    const walletsQuoteUnit = React.useMemo(() =>
+        getWallets(wallets, marketsQuoteUnit), [wallets, marketsQuoteUnit]);
```

Comment:
This definitely should be changed. `useMemo` will call `getWallets` or `getCurrencyFiltred` but will not re-render component. Still we have here performance issue.
You can find details here: https://reactjs.org/docs/hooks-reference.html#usememo
Replace it with `useCallback `